Applications/Utils/ModelPreparation/PartitionMesh/NodeWiseMeshPartitioner.cpp

```diff
@@ -262,6 +333,133 @@ NodeWiseMeshPartitioner::getNumberOfIntegerVariablesOfElements(
+                pvmd.is_data_type_signed = false;
+                pvmd.data_type_size_in_bytes = sizeof(double);
+                pvmd.number_of_components = pv->getNumberOfComponents();
+                pvmd.number_of_tuples = pv->getNumberOfTuples();
+                writePropertyVectorValuesBinary(out_val, *pv);
+            }
+        }
+        {
+            if (properties.existsPropertyVector<float>(name))
+            {
+                auto* pv = properties.getPropertyVector<float>(name);
+                pvmd.is_int_type = false;
+                pvmd.is_data_type_signed = false;
+                pvmd.data_type_size_in_bytes = sizeof(float);
+                pvmd.number_of_components = pv->getNumberOfComponents();
+                pvmd.number_of_tuples = pv->getNumberOfTuples();
+                writePropertyVectorValuesBinary(out_val, *pv);
+            }
+        }
+        {
+            if (properties.existsPropertyVector<int>(name))
+            {
+                auto* pv = properties.getPropertyVector<int>(name);
+                pvmd.is_int_type = true;
+                pvmd.is_data_type_signed = true;
+                pvmd.data_type_size_in_bytes = sizeof(int);
+                pvmd.number_of_components = pv->getNumberOfComponents();
+                pvmd.number_of_tuples = pv->getNumberOfTuples();
+                writePropertyVectorValuesBinary(out_val, *pv);
+            }
+        }
+        {
+            if (properties.existsPropertyVector<unsigned>(name))
+            {
+                auto* pv = properties.getPropertyVector<unsigned>(name);
+                pvmd.is_int_type = true;
+                pvmd.is_data_type_signed = false;
+                pvmd.data_type_size_in_bytes = sizeof(unsigned);
+                pvmd.number_of_components = pv->getNumberOfComponents();
+                pvmd.number_of_tuples = pv->getNumberOfTuples();
+                writePropertyVectorValuesBinary(out_val, *pv);
+            }
+        }
+        MeshLib::IO::writePropertyVectorMetaDataBinary(out, pvmd);
+    }
+    unsigned long offset = 0;
+    for (const auto& partition : _partitions)
+    {
+        MeshLib::IO::PropertyVectorPartitionMetaData pvpmd;
+        pvpmd.offset = offset;
+        pvpmd.number_of_tuples = partition.nodes.size();
+        INFO(
+            "Write meta data for PropertyVector: global offset %d, number "
+            "of tuples %d",
+            pvpmd.offset, pvpmd.number_of_tuples);
+        MeshLib::IO::writePropertyVectorPartitionMetaData(out, pvpmd);
+        offset += pvpmd.number_of_tuples;
+    }
+    out.close();
+    out_val.close();
+}
+
+void NodeWiseMeshPartitioner::readPropertiesConfigDataBinary(
+    const std::string& file_name_base) const
+{
+    const std::string fname = file_name_base + "_partitioned_properties_cfg"
+                              + std::to_string(_npartitions) + ".bin";
+    std::ifstream is(fname.c_str(), std::ios::binary | std::ios::in);
+    if (!is)
```

Comment:
Since the parallel reading is implemented by your own instead of using MPI_File_read, you need more MPI barriers to synchronise the computations among compute code.  Besides, if conditions, which may be in different status on different compute cores,  have be to check by MPI communication. Therefore,  I am not sure about the efficiency of the current implementation.

You can keep the current implementation by adding
``MPI_Barrier(_mpi_comm )`` before this if-condition and before the end of this function.